cf-keycrypt/cf-keycrypt.c

```diff
@@ -0,0 +1,388 @@
+    }else if (!(infile = fopen(cryptfile, "r"))) {
+        fprintf(stderr, "Error: Cannot locate input file.\n");
+        return -1;
+    }
+
+    if ((strcmp("-",plainfile) == 0)) {
+        outfile = stdout;
+        //printf("Write to stdout\n");
+    }else if (!(outfile = fopen(plainfile, "w"))){
+        fprintf(stderr, "Error: Cannot create output file.\n");
+        return -1;
+    }
+
+    ks = RSA_size(key);
+    tmpciph = (unsigned char *)malloc(ks * sizeof(unsigned char));
+    tmpplain = (unsigned char *)malloc(ks * sizeof(unsigned char));
+    //printf("Keysize: %d\n", ks);
+    while(!feof(infile))
+    {
+        memset(tmpciph, '\0', ks);
+        memset(tmpplain, '\0', ks);
+        len = fread(tmpciph, 1, ks, infile);
+        if (len > 0){
+            if ((size = RSA_private_decrypt(ks, tmpciph, tmpplain, key, RSA_PKCS1_PADDING)) == -1)
+            {
+                fprintf(stderr, "%s\n", ERR_error_string(ERR_get_error(), NULL));
+                return -1;
+            }
+        }
+        fwrite(tmpplain,1,strlen(tmpplain),outfile);
+    }
+    fclose(infile);
+    fclose(outfile);
+    free(tmpplain);
+    free(tmpciph);
+    RSA_free(key);
+    return plsz;
+}
+
+int main(int argc, char *argv[])
+{
+    OpenSSL_add_all_algorithms();
+    ERR_load_crypto_strings();
+
+    opterr = 0;
+    char *key = NULL;
+    char *infile = NULL;
+    char *outfile = NULL;
+    char *host = NULL;
+    int encrypt = 0;
+    int decrypt = 0;
+    int c = 0;
+    int size = 0;
+
+    //strcpy(WORKDIR,CFCR_WORKDIR);
+
+    while ((c = getopt (argc, argv, "he:d:i:o:H:")) != -1)
+    {
+        switch (c)
+        {
+            case 'e':
+                encrypt = 1;
+                key = optarg;
+                break;
+            case 'd':
+                decrypt = 1;
+                key = optarg;
+                break;
+            case 'i':
+                infile = optarg;
+                break;
+            case 'o':
+                outfile = optarg;
+                break;
+            case 'H':
+                encrypt = 1;
+                host = optarg;
+                break;
+            case 'h':
+                usage();
+                exit(1);
+            default:
+                printf("ERROR: Unknown option '-%c'\n", optopt);
+                usage();
+                exit(1);
+        }
+    }
+    if (host)
+    {
+        key = get_host_pubkey(host);
+        if (!key)
+        {
+            fprintf(stderr,"ERROR: Unable to locate public key for host %s\n", host);
+            exit(1);
+        }
+    }
+
+    if (encrypt > 0 && key && infile && outfile)
+    {
+        size = rsa_encrypt(key, infile, outfile);
+        if (size < 0)
+            exit(1);
+    }
+    else if (decrypt > 0 && key && infile && outfile)
+    {
+        size = rsa_decrypt(key, infile, outfile);
+        if (size < 0)
+            exit(1);
+    }
+    else
+    {
+        usage();
+        exit(1);
+    }
+    exit(0);
```

Comment:
Replace all zero and ones with EXIT_SUCCESS and EXIT_FAILURE. I would also suggest to just use `return retval` just at the end, if it's easy.

Reply:
Fixed.